### PR TITLE
Split framework struct

### PR DESCRIPTION
**Pull request by Suresh-Nakkeran**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

related to #1924 

**Special notes for your reviewer**:

1. Rename `Framework` to `ModelFormat` on `InferenceService` CR
2. Rename `Framework` to `SupportedModelFormat` on `ClusterServingRuntime` CR as it can contain more fields like `AutoSelect` configuration flags. 

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/apis/serving/v1alpha1/servingruntime_types.go [hunk 1]
```diff
@@ -21,7 +21,8 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type Framework struct {
+// +k8s:openapi-gen=true
+type ServingRuntimeFramework struct {
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/servingruntime_types.go]: Per slack discussion we'd like to rename this to `SupportedModelFormat` as currently we use a mixed list of model frameworks like `tensorflow`, `pytorch`, `sklearn`, `xgboost` and model formats `onnx`, `pmml`, `torchscript`, `tensorrt`, so `ModelFormat` is more inclusive.

File: pkg/apis/serving/v1alpha1/servingruntime_types.go [hunk 2]
```diff
@@ -88,9 +92,10 @@ type ServingRuntimePodSpec struct {
 // ServingRuntimeSpec defines the desired state of ServingRuntime. This spec is currently provisional
 // and are subject to change as details regarding single-model serving and multi-model serving
 // are hammered out.
+// +k8s:openapi-gen=true
 type ServingRuntimeSpec struct {
 	// Model formats and version supported by this runtime
-	SupportedModelTypes []Framework `json:"supportedModelTypes,omitempty"`
+	SupportedModelTypes []ServingRuntimeFramework `json:"supportedModelTypes,omitempty"`
```
Comment (yuzisun) [inline pkg/apis/serving/v1alpha1/servingruntime_types.go]: `SupportedModelTypes` -> `SupportedModelFormats` 
`ServingRuntimeFramework` -> `SupportedModelFormat`

File: pkg/apis/serving/v1beta1/predictor_model.go [hunk 3]
```diff
@@ -27,10 +27,21 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type Framework struct {
```
Comment (yuzisun) [inline pkg/apis/serving/v1beta1/predictor_model.go]: Rename Framework to ModelFormat

**Comment by Suresh-Nakkeran** [comment]
@pvaneck @yuzisun I updated the PR with recommended struct names. Can you please review the changes?

**Comment by yuzisun** [comment]
thanks @Suresh-Nakkeran !

/lgtm
/approve

**Pull request merged**